frontend/components/collab/code-editor.tsx

```diff
@@ -0,0 +1,149 @@
+    language: "cpp",
+    value: "// some comment",
+  },
+};
+
+export default function CodeEditor({ roomId }: { roomId: string }) {
+  const monaco = useMonaco();
+  const [language, setLanguage] = useState<string>("Javascript");
+  const [theme, setTheme] = useState<string>("light");
+  const ydoc = useMemo(() => new Y.Doc(), []);
+  const [editor, setEditor] =
+    useState<MonacoEditor.IStandaloneCodeEditor | null>(null);
+  const [provider, setProvider] = useState<WebsocketProvider | null>(null);
+  const languageMap = useMemo(() => ydoc.getMap("language"), [ydoc]);
+
+  useEffect(() => {
+    const provider = new WebsocketProvider(
+      "ws://localhost:3002/yjs",
+      roomId,
+      ydoc
+    );
+    setProvider(provider);
+    return () => {
+      provider?.destroy();
+      ydoc.destroy();
+    };
+  }, [ydoc, roomId]);
+
+  useEffect(() => {
+    if (provider == null || editor == null) return;
+    const monacoBinding = new MonacoBinding(
+      ydoc.getText(),
+      editor.getModel()!,
+      new Set([editor]),
+      provider?.awareness
+    );
+    return () => {
+      monacoBinding.destroy();
+    };
+  }, [ydoc, provider, editor]);
+
+  useEffect(() => {
+    const handleLanguageChange = () => {
+      const newLanguage = languageMap.get("selectedLanguage") as string;
+      if (newLanguage && newLanguage !== language) {
+        setLanguage(newLanguage);
+      }
+    };
+    languageMap.set("selectedLanguage", language);
+    languageMap.observe(handleLanguageChange);
+    return () => {
+      languageMap.unobserve(handleLanguageChange);
+    };
+  }, [languageMap, language]);
+
+  useEffect(() => {
+    if (editor && monaco) {
+      const model = editor.getModel();
+      if (model) {
+        monaco.editor.setModelLanguage(model, languages[language]?.language);
+      }
+    }
+  }, [language, editor, monaco]);
+
+  useEffect(() => {
+    if (monaco) {
+      monaco.editor.setTheme(theme === "dark" ? "vs-dark" : "light");
+    }
+  }, [theme, monaco]);
+
+  return (
+    <div className="w-3/5 p-4">
+      <Card className="h-full flex flex-col">
+        <CardHeader>
+          <CardTitle>Code Editor</CardTitle>
+        </CardHeader>
+        <CardContent className="flex-1 flex flex-col">
+          <div className="h-1/8 mb-4 flex justify-between items-center">
+            <Select value={language} onValueChange={setLanguage}>
+              <SelectTrigger className="w-1/5">
+                <SelectValue placeholder="Select Language" />
+              </SelectTrigger>
+              <SelectContent>
+                {Object.keys(languages).map((lang) => (
+                  <SelectItem key={lang} value={lang}>
+                    {lang}
+                  </SelectItem>
+                ))}
+              </SelectContent>
+            </Select>
+            <Button
+              onClick={() => setTheme(theme === "light" ? "dark" : "light")}
+              variant={theme === "light" ? "secondary" : "default"}
```

Comment:
Would be nice if we could use the theme provider to get the theme instead of having a separate light/dark mode

Reply:
Fix would be to use `useTheme` from the provider

Reply:
Resolved in chat, more flexibility for users and we can support more themes in the future